woocommerce/woocommerce-gutenberg-products-block
# Add Product Query Support for Atomic Rating Block

Adds Product Query support for the atomic Rating block.

On the client side, when the Rating Block is used within the Product Query block, the markup will be rendered on the server side - no javascript related to the Rating block will be loaded.





Fixes #7330 





### Testing

1. Checkout this branch and run `npm start` (need to run the `start` command to catch the experimental flags).
1. Be sure that you have the latest version of the Gutenberg feature plugin installed and activated.
1. Add the `woocommerce/product-rating` block to the Product Query `INNER_BLOCKS_TEMPLATE` in `assets/js/blocks/product-query/constants.ts` under line 54 (**only for testing, remove afterwards**).
1. Choose at least one product and add a rating to it (this can be done on a single product view for any product).
1. Create a post/page and add the Product Query block.
1. Save it.
1. Open the frontend side and confirm that:
    - No JS related to the Product Rating block is loaded (search in your network tab for `product-rating-frontend.js`).
    - The UI of the block is identical to on the editor side.
    - That the settings are reflected correctly (eg: custom style).
1. Confirm that the rating stars for the product that you've added the rating to appear as expected in the UI.
1. Back in the editor confirm that the Product Rating block can be added as a child of the Product query block via the inserter.
1. For comparison, switch to trunk and add the All Products block to confirm that the aforementioned `product-rating-frontend.js` _is_ loaded.


### WooCommerce Visibility
